### DISCUSSION TITLE
Change of Default Branch Name

### DISCUSSION THREAD

**Discussion by Flyingmana** [tenacityteam/tenacity-legacy#167] [General]
I know its probably not a good time to open up another potential sensitive topic, but on the other side we are still very much in the beginning, and changing the default branch is still easy to do.

Most git hosting platforms did already move for new projects to `main` as default branch, including Github.

https://github.blog/changelog/2020-10-01-the-default-branch-for-newly-created-repositories-is-now-main/

You can read up more informations about this here: https://github.com/github/renaming/

**Comment by falkTX** [comment]
Removing one branch on github will auto-close all PRs that target that branch.
While I am mostly positive about the change, it needs to happen together with cleaning up the existing PRs, all at once.

**Reply by Flyingmana** [reply]
It could also be done in steps.
1. creating the new branch, and change to be the default
2.  
    a) pull regularly the master into the main (till all branches were merged)
    b) change the target on all PRs to main
3. remove master

**Reply by Be-ing** [reply]
Yeah it is annoying that GitHub does that. I learned that the hard way before. The new branch could be created first, existing PRs can manually be retargeted, then delete the old branch.

**Reply by Sciss** [reply]
At this point, number of open PRs is also low - so might be a good moment.

**Reply by fossdd** [reply]
GitHub has a button for renaming the default branch. https://github.com/tenacityteam/tenacity/settings/branches. Does GitHub renames the target branches with that? I don't know.

**Reply by Songtech-0912** [reply]
No - as per [this SO answer](https://stackoverflow.com/questions/57570830/does-renaming-a-github-organization-affect-the-forks-of-that-organizations-proj), local branches will need to update their target branch if a branch rename occurs.

**Reply by fossdd** [reply]
Oh, then I think closing the all PRs is not good and will never be an option. Maybe if tenacity **fully** (1) migrated to sourcehut the team can change it.

**Comment by objectifnul** [comment]
By the way, there is at least one other github project with the same name (https://github.com/jd/tenacity)

**Reply by Be-ing** [reply]
No matter what name you come up with, someone is probably already using it for something. The question is whether people are likely to confuse them. That software you linked has nothing to do with audio, so I am not concerned.

**Comment by Songtech-0912** [comment]
I agree with the change, it is a positive one. However, let's first get communication and project governance right, because then we have a unified way to communicate this change without causing a lot of knock-on effects downstream (e.g. for contributions). Once we are ready we can signal a temporary halt to all contributions as this change is being made.

**Comment by ErikPrantare** [comment]
The word `master` in this context has no relation to slavery as far as I know, which is the argument being pushed as the reason to switch names.
I have no strong feelings on which name to use, just find it curious that Americans seem so eager to search for problematic language where there frankly is none. It looks a tad bit religious for an outsider.

**Reply by Sciss** [reply]
Please don't do this. This topic of the origin and meaning or connotation of the default branch has been discussed hundred times across all sorts of projects. We don't need to repeat that discussion for the 101st time.

**Reply by ErikPrantare** [reply]
Could you link to a discussion of the origin and meaning of `master` in this context? To me it is more connected to the use of a "master" copy.

**Reply by Songtech-0912** [reply]
The idea of "master" is that of a "master copy" here, we all get that. But please understand that the master-slave analogy is used or had been used extensively in the FOSS community, and we recognize that while it might not have been bad intentioned, that doesn't mean it isn't hurtful towards certain underrepresented groups, and it *isn't* something to be proud of.

**Reply by Sciss** [reply]
as a conciliating fact, even if not everyone agrees that it is problematic, there is technically no problem of renaming to `main`; it's a one-off change, after which procedures work precisely as before (except you need to type two characters less). The SuperCollider project did that change recently, after a winded discussion with the same arguments (US cultural imperialism etc). The change was quickly accepted, people ran `set-url`, nobody shed a tear.

**Reply by ErikPrantare** [reply]
I'm all for creating a welcoming space. But the reason for switching from "master" to "main" is that github (i.e. micro$oft) did it to score brownie points during the George Floyd protests in the US. To me, this looks like part of the broader infestation of all areas of living with corporate ideology. Agree with silicon valley, or be shunned. I don't think these changes have any meaningful positive material impact in anybody's life. I don't really care about the name of the branch if you would've taken it in a vacuum. Rather, it's the fact that HR culture is seeping into large areas of the Free Software community. Does it not bother you that many of these issues are pushed by the enforcers of bourgeois ideology?

**Reply by falkTX** [reply]
huh so how do you explain gitlab doing the same?
and many of the linux kernel code replacing master/slave with other terms (primary/secondary, host/device, etc).

**Reply by falkTX** [reply]
example of something quick I found: https://lore.kernel.org/lkml/20200630174123.GA1906678@kroah.com/T/

is there some kind of hidden agenda here too? :thinking:

**Reply by ErikPrantare** [reply]
It's not not a hidden agenda. It's just the hegemonic ideology of the west. Myopically focused on minute details to signal inclusivity, yet almost always completely devoid of any real class consciousness.

Since when are the linux kernel developers against corporate domination? They implemented enforcement of Digital Restrictions Management in the kernel (https://techrights.org/2019/09/26/linux-as-open-source-proprietary-software/).

**Reply by Songtech-0912** [reply]
@ErikPrantare We all know that the Linux kernel and its developers have a mostly laissez-faire attitude towards corporate use. However, I'm pretty sure that the name changes are not because of ideological concerns, but rather because of an emerging idea that it's the right thing to do. DRM in the kernel, that's not the best of things, but without it the kernel would be unusable on many machines (due to hardware considerations). That's a separate topic though.

**Reply by falkTX** [reply]
As I can see, it is just being nice.
It doesn't hurt us and it is not too much of a pain, so why not.
We are not idiots you know..

PS: techrights.org have continuously sided with freenode, hard to take links from them seriously.

**Reply by ErikPrantare** [reply]
@Songtech-0912 
I respect your view on this. I recognize that there isn't one single reason that people are for the name change, and that many genuinely think that it would make a difference. As I said, I don't have an issue with the name "main", so I'll be fine with it being changed. I just think it's concerning that the US "left" seems so blind to the dynamic between them and corporate. As it stands, the American left seems to be able to go together quite well with corporations, media conglomerates, and their politicians. See e.g. how many cheered on the fusion of politics and corporate power after the attempted january insurrection, where they actively censored a *former US president*. This is getting off-topic however, so I'll leave it at that.

@falkTX
I don't think you are an idiot. In fact, I'm 100% sure that you're doing this because you think it's a nice thing to do.
And idc about what techrights have said in other contexts. They don't become untouchable simply for being wrong on one particular point.

**Reply by Songtech-0912** [reply]
@ErikPrantare No issue, I also respect your view on the topic. However, let's just get this done and over with. Don't let it fester and cause divides in the community.

**Comment by Songtech-0912** [comment]
Please, please, let's not start a culture war here. TLDR is that **you can change master-slave terminology regardless of your position and background**. Removing master/slave names are NOT political positions or signs of corporate infiltration. Let's make this clear:

- The ideas of using master-slave terminology are simply outdated. Just as we do not say "all men are equal" today, instead we say "all human beings are equal". That terminology was from a past time, and now we realize that it's not the best term to use, and so we replace it with a newer name. I think that's practically universally acceptable?
- Changing the branch name to main DOES NOT constitute an ideological push towards something:
   - To be sure, I hate the fact that some companies do things because they want to make it like they care but don't (e.g. Google's marketing around the Chrome browser...*"Chrome works hard to protect your data and privacy online"* absolute lie...)
   - However, I'm pretty sure that even though that's the reason why some companies did the branch name change, most were well-intentioned
- Talking about "enforcers of bourgeois ideology" - I want to be lenient here, but please not go into politics? The Contributor Covenant AND the GNU Kind Communications Guidelines explicitly recommend that we **don't go into political matters** in a community.
- I think we can agree, regardless of origin/ideology/philosophy/physiology that basic human rights are something we should all care about. And "master slave" terminology directly conflicts with those basic human rights.
- Also trying to insist on keeping an old term in use way beyond its point of use breeds an unwelcome, unkind spirit. We want to build a *welcoming* community. Let's just be nice (as others have repeated before).

**Reply by ErikPrantare** [reply]
I generally agree with the spirit of what you write. I think the reason these discussions become political is because the master/main and CoC discussions are inherently political in nature, even if they might not seem like it for someone who agrees. I didn't find my comments about HR to be political, but I understand that the choice of words signals political affiliation which makes it seem so. I think the same is true about master/main and CC.

I also haven't heard people say "all human beings are equal", but I'm not an Anglo so maybe I've missed something :^)

As I said above, it's not the master->main itself that concerns me, so feel free to change it if you feel it is worth the small effort. I just hope more people become more aware of corporate influence over FS (just look at this discussion about it taking place on a M$ site), and try to take a more active stance against them even if they are using your politically preferred language. It doesn't matter in this specific situation, but it surely will some other time (imo it has already happened in a recent case, but discussing that here will most likely just derail the conversation).

**Reply by Songtech-0912** [reply]
Ok got it. Disagreements in any community are ok and perfectly acceptable. However, we will definitely take an active stance against corporate influence - we literally "created" Tenacity to defend our ability to use software without corporate influence, and all of the community agrees with that. We'll continue to do so as much as we can.
